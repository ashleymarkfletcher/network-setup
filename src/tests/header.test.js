import React from 'react';
import renderer from 'react-test-renderer';
import Header from '../components/header';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

test('Header renders', () => {

  const mockActiveInterface = {
    gateway_ip: "192.168.20.1",
    ip_address: "192.168.20.106",
    mac_address: "34:13:E8:29:47:B2",
    model: "Intel(R) Dual Band Wireless-AC 7265",
    name: "Wi-Fi",
    netmask: "255.255.255.0",
    type: "Wireless",
    vendor: "Intel Corporation"
  }

  const mockCurrentInterface = {
    gateway_ip: "192.168.20.1",
    ip_address: "192.168.20.106",
    mac_address: "34:13:E8:29:47:B2",
    model: "Intel(R) Dual Band Wireless-AC 7265",
    name: "Wi-Fi",
    netmask: "255.255.255.0",
    type: "Wireless",
    vendor: "Intel Corporation"
  }

  const mockInterfaces = [
    {
      gateway_ip: "192.168.20.1",
      ip_address: "192.168.20.106",
      mac_address: "34:13:E8:29:47:B2",
      model: "Intel(R) Dual Band Wireless-AC 7265",
      name: "Wi-Fi",
      netmask: "255.255.255.0",
      type: "Wireless",
      vendor: "Intel Corporation"
    },
    {
      gateway_ip: null,
      mac_address: "B8:AE:ED:75:14:22",
      model: "Intel(R) Ethernet Connection (3) I218-V",
      name: "Ethernet",
      type: "Wired",
      vendor: "Intel Corporation"
    }
  ]

  const close = () => {}

  const updateInterface = () => {}

  const tree = shallow(
      <Header
        close={close}
        activeInterface={mockActiveInterface}
        interfaces={mockInterfaces}
        updateInterface={updateInterface}
        currentInterface={mockCurrentInterface}
      >
      </Header>
  )

  expect(shallowToJson(tree)).toMatchSnapshot();
})
