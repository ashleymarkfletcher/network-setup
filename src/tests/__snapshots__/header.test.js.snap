// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders 1`] = `
<div
  className="header"
  style={
    Object {
      "paddingBottom": 5,
      "paddingLeft": 20,
      "paddingTop": 5,
    }
  }
>
  <div
    className="currentInterface"
  >
    <p>
      Active Interface: 
      Wi-Fi
    </p>
  </div>
  <div
    className="selectInterface"
  >
    <p>
      Select Interface:
    </p>
    <DropDownMenu
      anchorOrigin={
        Object {
          "horizontal": "left",
          "vertical": "top",
        }
      }
      animated={true}
      autoWidth={true}
      disabled={false}
      iconButton={<NavigationArrowDropDown />}
      labelStyle={
        Object {
          "color": "white",
        }
      }
      maxHeight={500}
      multiple={false}
      onChange={[Function]}
      openImmediately={false}
      style={Object {}}
      value={
        Object {
          "gateway_ip": "192.168.20.1",
          "ip_address": "192.168.20.106",
          "mac_address": "34:13:E8:29:47:B2",
          "model": "Intel(R) Dual Band Wireless-AC 7265",
          "name": "Wi-Fi",
          "netmask": "255.255.255.0",
          "type": "Wireless",
          "vendor": "Intel Corporation",
        }
      }
    >
      <MenuItem
        anchorOrigin={
          Object {
            "horizontal": "right",
            "vertical": "top",
          }
        }
        checked={false}
        desktop={false}
        disabled={false}
        focusState="none"
        insetChildren={false}
        key="Wi-Fi"
        primaryText="Wi-Fi"
        targetOrigin={
          Object {
            "horizontal": "left",
            "vertical": "top",
          }
        }
        value="Wi-Fi"
      />
      <MenuItem
        anchorOrigin={
          Object {
            "horizontal": "right",
            "vertical": "top",
          }
        }
        checked={false}
        desktop={false}
        disabled={false}
        focusState="none"
        insetChildren={false}
        key="Ethernet"
        primaryText="Ethernet"
        targetOrigin={
          Object {
            "horizontal": "left",
            "vertical": "top",
          }
        }
        value="Ethernet"
      />
    </DropDownMenu>
  </div>
</div>
`;
